ops/model.py

```diff
@@ -3413,6 +3426,10 @@ def status_set(self, status: str, message: str = '', *, is_app: bool = False) ->
         """
         if not isinstance(is_app, bool):
             raise TypeError('is_app parameter must be boolean')
+        if not isinstance(message, str):
+            raise TypeError('message parameter must be a string')
+        if status not in _SettableStatusNames:
+            raise ModelError(f'status must be in {_SettableStatusNames}, not {status}')
```

Comment:
Maybe this should be `InvalidStatusError` rather than `ModelError`?

Also, while I'm here - I noticed when looking over the docs that we say that `ModelError` will be raised if using `ErrorStatus` or `UnknownStatus` to set - could you fix that to say `InvalidStatusError` (if I'm correct in thinking that is the case - it certainly seems like it from the rest of the docs).

Reply:
+1 on InvalidStatusError from me.

While this could (pedantically) be considered a breaking change at the API level (someone may have called this function with `status="error"` and got away with it), it seems that none's doing that, according to super-tox. Arguably it wouldn't be very useful due to juju hook tool behaviour either.

Pls grep known charm code for `"unknown"` and `"error"` before resolving.

Reply:
P.S. given that it's a string, wdyt about:

```suggestion
            raise ModelError(f'status must be in {_SettableStatusNames}, not {status!r}')
```

Reply:
I agree on changing to `InvalidStatusError`, especially as it's a subclass of `ModelError`, so it's unlikely to break anybody's `try ... except ...`

Anything in particular you'd like me to grep for? unknown and error are such generic terms that they're widely used in many codebases. I did find one interesting result in namespace-node-affinity-operator, where `unit.status` is set to `ErrorStatus` on [this line](https://github.com/canonical/namespace-node-affinity-operator/blob/3b24b7c166461c77ff313400d5089e8136099cae/src/charm.py#L187):
```
            self.model.unit.status = ErrorStatus("K8S resources removal failed")
```
This will currently cause the charm to raise an unhandled `ModelError`, and with this PR would instead cause it to raise an unhandled `InvalidStatusError`.